### ISSUE TITLE
A "management" image/container

### ISSUE THREAD

**Issue by prologic** [adaline/dockermail#16]
This might be combined with the "setup" image/container in some way; but I'm thinking of extending what I've done with the "setup" once-off image to build tooling that does:
```#!bash
docker-compose run setup add <email> <password>
docker-compose run setup add <email> <password> [<alias>, ...]
docker-compose run setup rm <email>
docker-compose run setup ls
```

Thoughts?

The "domain" would be implied by the ``<email``. I do not yet have a PR to this effect; but this is what I'm thinking of building.

**Comment by prologic** [comment]
Got the ``add`` part working :)
```#!bash
$ docker-compose run setup add user@domain.com
Enter password for user@domain.com:
$ docker-compose restart core
```

New account created and can send email to and login via IMAP and Rainloop.

**Comment by adaline** [comment]
That is an excellent idea, would really help first time setup.
My thinking with images was to bundle things you need to set it up into the image itself.
So if you want to set up core, the setup script is in the core itself etc.
I think this way there is no need to modify a separate "setup" image every-time one module changes.
What do you think on that?
Once again, thank you so much for all the feedback and work, i was not expecting anyone to feel interested in this project so its amazing to hear form you :+1:

**Comment by prologic** [comment]
I think the way I've got this working is quite nice. It *doesn't* involve "modifying" a setup image however. It's just a "package" set of tooling to manage the configuration. I now have:
```#!bash
docker-compose run setup  # initial setup
docker-compose run setup add <email>  # adding a new user/domain
```

**Comment by Lednerb** [comment]
@adaline This Project is such a great thing!

@prologic Very nice work! I'm looking forward to see the other parts of the setup package working.

I'm trying to use this project in an production environment for some clients. The only problem I see so far is that maybe there's an update-problem when a user changed his mail password?

To update the mail server i just have to re-run the `docker-compose up -d` command, right?
But than all changed passwords would be overwritten with the values in the `config.json` file?

Maybe it's possible to configure the core container that the mail passwords are saved hashed in the `dockermail/mail` or `dockermail/settings` folder?

Any ideas to fix this?

**Comment by adaline** [comment]
Hey @Lednerb,
Glad you like this project!
Im not sure how you change the password outside of updating `config.json`, manually or through the setup image/external tool.
In my case, I update the password hash in the `config.json` and restart the containers, which will
update internal config.

**Comment by Lednerb** [comment]
You're right. Just tried to change the password via rainloop webmail client.
The client displays a success message but indeed the password does not change.

_maybe off-topic:_

I want to integrate the password changing functionality in my custom php-backend so the clients can access their dashboard and can configure their mail settings. But thats not the "docker way" of doing this because an application within a docker container would influence another container... At the moment i also have not really an idea how to restart dockermail when the config.json has changed automatically...

**Comment by adaline** [comment]
@Lednerb,
you can add a script to watch config.json and restart dockermail when it changes.
See this page for a good starting point: http://superuser.com/questions/181517/how-to-execute-a-command-whenever-a-file-changes
